purchase_order_product_recommendation_secondary_unit/wizards/purchase_order_recommendation.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 David Vidal <david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from odoo.tools.float_utils import float_compare, float_round
+from math import ceil
+
+
+class PurchaseOrderRecommendation(models.TransientModel):
+    _inherit = 'purchase.order.recommendation'
+
+    @api.model
+    def _add_product_values(self, product_ids, found_dict):
+        # Compute and round packs up
+        res = super()._add_product_values(product_ids, found_dict)
+        for product in product_ids.filtered('purchase_secondary_uom_id'):
+            line = found_dict[product.id]
+            line['secondary_uom_id'] = product.purchase_secondary_uom_id.id,
```

Comment:
Isn't simpler to add a hook method for preparing line values that avoids to loop again on all lines?

Reply:
Well, I have it. I should add the product_id record on every dict key that's true.

Reply:
Solved